src/Service/Request.service.js

```diff
@@ -41,18 +41,47 @@ export default class RequestService extends DependencyAwareClass {
     return typeof queryParameters[parameter] !== 'undefined' ? queryParameters[parameter] : ifNull;
   }
 
+  /**
+   * Get all HTTP headers included in the request.
+   *
+   * @returns {object} An object with a key for each header.
+   */
+  getAllHeaders() {
+    return { ...this.getContainer().getEvent().headers };
+  }
+
+  /**
+   * Get an HTTP header from the request.
+   *
+   * The header name is case-insensitive.
+   *
+   * @param {string} name The name of the header.
+   * @param {string} [whenMissing] Value to return if the header is missing.
+   *   (default: empty string)
+   *
+   * @returns {string}
+   */
+  getHeader(name: string, whenMissing: string = '') {
+    const { headers } = this.getContainer().getEvent();
```

Comment:
Do you think we could call `getAllHeaders()` here?

It is unlikely we will ever refactor where the headers are but if we do we will have only one place to update

Reply:
Yes, good point! 09ad78c